feat: migrate old OpenAPI implementation to Loader interface

bzp2010 opened the pull request:
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [ ] New feature provided
- [x] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Migrate legacy code to the Loader interface implementation to decouple the handler from the import and export logic

**Related issues**

a part of #2465 

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [x] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

bzp2010 commented:
### Update

The original OpenAPI import and export logic has been migrated to the Loader interface implementation, including the associated test cases.
However, it should be noted that only code migration has been done in this PR, not removing the original coupling with the Handler, which is still there; the Handler is too coupled and will be completely rewritten later to adapt to the implementation of the Loader interface.

### Desc

The migration in this PR is consistent with the original code, with no changes to the overall logic and only minor adjustments to adapt the Loader interface; the test cases have also been modified to allow testing the import and export modules separately without including logic at the level of the HTTP application framework (Gin). And, once a huge test case file with 2000 lines of code was split into multiple code files by module.

**Therefore, in this PR, the import and export logic coupled with the Gin handler is split out and provided as a separate loader package.**

nic-chen commented:
large PR, maybe we could split it into small PRs? for example, at least two, import and export.

starsz commented:
> large PR, maybe we could split it into small PRs? for example, at least two, import and export.

Agree +1

bzp2010 commented:
> large PR, maybe we could split it into small PRs? for example, at least two, import and export.

Hi, @nic-chen.

This one PR contains only the old code and tests for the migration, which are essentially replicated with no core modification logic and only a few modifications for adaptation reasons.

This PR has a huge number of lines, mainly test cases, the original test cases are up to 2000 lines, I split them into several test files for different purposes; the core code has about 500 lines, only a little traversal code is added in the function entry part; the remaining part is the license header.

So I think it's important to focus only on the part I added, which is [codes](https://github.com/apache/apisix-dashboard/blob/ad888d0753726b017335090c94113651b1d55cc4/api/internal/handler/data_loader/loader/openapi_legacy/export.go#L51-L68); and no need to split the PR.

bzp2010 commented:
### Update

#### I'm even thinking about whether it's necessary for it to continue to exist, it only exists to ensure partial compatibility, and given its poor implementation, it's difficult for me to devote effort to maintaining it (The logic of the code is confusing, containing various global and local variables, nested loops; the tests are even run completely on mocking, and the test data they provide is not executed correctly in real situations at all.), and this old implementation will probably be removed later (when we have a better implementation).
#### This PR is currently low priority, and I will deal with this at the end if the current PR is not available for review.

cc @nic-chen @starsz

starsz commented:
> ### Update
> #### I'm even thinking about whether it's necessary for it to continue to exist, it only exists to ensure partial compatibility, and given its poor implementation, it's difficult for me to devote effort to maintaining it (The logic of the code is confusing, containing various global and local variables, nested loops; the tests are even run completely on mocking, and the test data they provide is not executed correctly in real situations at all.), and this old implementation will probably be removed later (when we have a better implementation).
> #### This PR is currently low priority, and I will deal with this at the end if the current PR is not available for review. Several other PRs have higher priority, such as #2460 #2474
> cc @nic-chen @starsz

Got it.
